features/users/models.py

```diff
@@ -1,13 +1,48 @@
-from sqlalchemy import String, LargeBinary
-from sqlalchemy.orm import Mapped, mapped_column
+import datetime
+from typing import Optional
+
+from sqlalchemy import String, LargeBinary, Column, Table, ForeignKey, DateTime, func, Integer
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 
 from features import DbBaseModel
 
 
 class User(DbBaseModel):
     __tablename__ = 'Users'
 
-    id: Mapped[int] = mapped_column(primary_key=True)
+    id: Mapped[int] = mapped_column(primary_key=True, init=False)
     username: Mapped[str] = mapped_column(String(30), unique=True)
     email: Mapped[str] = mapped_column(String(50), unique=True)
     password: Mapped[bytes] = mapped_column(LargeBinary())
+
+    roles = relationship('Role', secondary='user_roles', back_populates='users', lazy='selectin')
+
+
+class Role(DbBaseModel):
+    __tablename__ = 'Roles'
+
+    id: Mapped[int] = mapped_column(primary_key=True, init=False)
+    name: Mapped[str] = mapped_column(String(50), unique=True)
+    created_on: Mapped[datetime.datetime] = mapped_column(DateTime, server_default=func.current_timestamp(), init=False)
+    created_by: Mapped[str] = mapped_column(String(30))
+
+    users = relationship('User', secondary='user_roles', back_populates='roles', lazy='selectin')
+
+
+user_roles = Table('user_roles', DbBaseModel.metadata,
```

Comment:
After a long fight to make it like other models, I asked chatgpt. Here is his answer:
"""
The issue may be related to the fact that you're using MappedAsDataclass in your DbBaseModel. The MappedAsDataclass mixin is designed to work with dataclasses, and when combined with declarative models in SQLAlchemy, it might be causing conflicts.
"""